presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -529,92 +505,70 @@ private void addToHiveSplitSourceRoundRobin(List<Iterator<InternalHiveSplit>> it
-                    }
-                    catch (IOException e) {
-                        throw Throwables.propagate(e);
-                    }
+        if (!splittable) {
+            // not splittable, use the hosts from the first block if it exists
+            blocks = ImmutableList.of(new InternalHiveBlock(start, start + length, blocks.get(0).getAddresses()));
+        }
 
-                    long targetChunkSize;
-                    if (remainingInitialSplits.decrementAndGet() >= 0) {
-                        targetChunkSize = maxInitialSplitSize.toBytes();
-                    }
-                    else {
-                        long maxBytes = maxSplitSize.toBytes();
-                        int chunks = toIntExact((long) Math.ceil((blockLocation.getLength() - chunkOffset) * 1.0 / maxBytes));
-                        targetChunkSize = (long) Math.ceil((blockLocation.getLength() - chunkOffset) * 1.0 / chunks);
-                    }
+        // TODO: remainingInitialSplits, maxInitialSplitSize, maxSplitSize
+        return Optional.of(new InternalHiveSplit(
+                partitionName,
+                path,
+                start,
+                start + length,
+                fileSize,
+                schema,
+                partitionKeys,
+                blocks,
+                bucketNumber,
+                splittable,
+                forceLocalScheduling && allBlocksHaveRealAddress(blocks),
+                columnCoercions));
+    }
 
-                    // adjust the actual chunk size to account for the overrun when chunks are slightly bigger than necessary (see above)
-                    long chunkLength = Math.min(targetChunkSize, blockLocation.getLength() - chunkOffset);
-
-                    InternalHiveSplit result = new InternalHiveSplit(
-                            partitionName,
-                            path,
-                            blockLocation.getOffset() + chunkOffset,
-                            chunkLength,
-                            fileSize,
-                            schema,
-                            partitionKeys,
-                            addresses,
-                            bucketNumber,
-                            forceLocalScheduling && hasRealAddress(addresses),
-                            columnCoercions);
-
-                    chunkOffset += chunkLength;
-
-                    if (chunkOffset >= blockLocation.getLength()) {
-                        checkState(chunkOffset == blockLocation.getLength(), "Error splitting blocks");
-                        blockLocationIterator.next();
-                        chunkOffset = 0;
-                    }
+    private static void checkBlocks(List<InternalHiveBlock> blocks, long start, long length)
+    {
+        checkArgument(!blocks.isEmpty());
+        checkArgument(start == blocks.get(0).getStart());
+        checkArgument(start + length == blocks.get(blocks.size() - 1).getEnd());
```

Comment:
Why doesn't this fail for the first block?

Reply:
I don't understand the question